### PR TITLE
Give dependency example for android clients.

### PR DESCRIPTION
**Pull request by madongfly**
Update README per the discussion in #1104

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,14 @@ Or for Gradle, add to your dependencies:
 compile 'io.grpc:grpc-all:0.9.0'
 ```
 
+For Android client, you only need to depend on the needed sub-projects, such as:
+```gradle
+compile 'io.grpc:grpc-core:0.9.0'
```
Comment (ejona86) [inline README.md]: Core should get pulled in via stub. Do we want to list it explicitly, or make this list minimal?

Comment (madongfly) [inline README.md]: Done, and also updated the example and interop test Apps.

**Comment by madongfly** [comment]
@ejona86 PTAL, thanks!

**Comment by ejona86** [comment]
@madongfly LGTM

**Pull request merged**